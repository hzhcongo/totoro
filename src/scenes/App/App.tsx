import React, { Component } from "react";
import logo from "resources/logo.svg";
import {
  BrowserRouter,
  Switch,
  Route,
} from "react-router-dom";

import "./App.css";
import { RouterButton } from "components/RouterButton";
import { MrtMap } from "scenes/MrtMap/MrtMap";

export class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <p>Totoro</p>
            <RouterButton
              id="takesATrainBtn"
              displayText="Takes a train"
              route="/MrtMap"
            />
            <RouterButton
              id="findsAHouseBtn"
              displayText="Finds a house"
              route="/MrtMap"
            />
          </header>
        </div>
        <Switch>
          <Route path="/MrtMap">
            <MrtMap />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

export default App;
