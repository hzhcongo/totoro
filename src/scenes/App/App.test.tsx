import React from "react";
import App from "./App";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import {
  createRenderer,
  setupTestEnvironment,
  tearDownTestEnvironment
} from "../../utilities/TestUtilities";

beforeAll(() => {
  setupTestEnvironment();
});

afterAll(() => {
  tearDownTestEnvironment();
});

describe("Snapshots", () => {
  test("Initial render", () => {
    let testRenderer: ReactTestRenderer = createRenderer(<App />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test("After train button clicked", () => {
    let testRenderer: ReactTestRenderer = createRenderer(<App />);

    const inst = testRenderer.root;
    const button = inst.findByProps({
      id:"takesATrainBtn",
      variant:"contained",
      color:"primary",
      size:"large"
		});
    renderer.act(() => {
      button.props.onClick();
    });
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
