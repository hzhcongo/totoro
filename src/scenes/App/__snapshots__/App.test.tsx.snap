// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots After train button clicked 1`] = `
Array [
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <p>
        Totoro
      </p>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        id="takesATrainBtn"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Takes a train
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        id="findsAHouseBtn"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Finds a house
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </header>
  </div>,
  <div
    className="MrtMap"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <p>
        Totoro
      </p>
    </header>
  </div>,
]
`;

exports[`Snapshots Initial render 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Totoro
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      id="takesATrainBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Takes a train
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      id="findsAHouseBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Finds a house
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </header>
</div>
`;
